python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -233,35 +233,15 @@ def owner(self) -> Any:
     @property
     def __cuda_array_interface__(self) -> Mapping:
         """Implementation of the CUDA Array Interface."""
-        return self._get_cuda_array_interface(readonly=False)
-
-    def _get_cuda_array_interface(self, readonly=False):
-        """Helper function to create a CUDA Array Interface.
-
-        Parameters
-        ----------
-        readonly : bool, default False
-            If True, returns a CUDA Array Interface with
-            readonly flag set to True.
-            If False, returns a CUDA Array Interface with
-            readonly flag set to False.
-
-        Returns
-        -------
-        dict
-        """
         return {
-            "data": (
-                self.get_ptr(mode="read" if readonly else "write"),
-                readonly,
-            ),
+            "data": (self.get_ptr(mode="write"), False),
             "shape": (self.size,),
             "strides": None,
             "typestr": "|u1",
             "version": 0,
         }
 
-    def get_ptr(self, *, mode) -> int:
+    def get_ptr(self, *, mode: Literal["read", "write"]) -> int:
```

Comment:
No need to change this, but just to note that `Literal` doesn't get handled very well by type-checkers if the argument comes from a variable, rather than a literal value (unless the variable is marked with `: Final`). Since they don't do dataflow analysis

https://mypy-play.net/?mypy=latest&python=3.11&gist=6541ee12e80daeb4b0837563d98dc442